### DISCUSSION TITLE
SeasonsInfo fields naming

### DISCUSSION THREAD

**Discussion by ebraminio** [cosinekitty/astronomy#190] [General]
Just wanted to share with you this library uses interesting naming for seasons, https://github.com/MenoData/Time4J/blob/78cd60d6/base/src/main/java/net/time4j/calendar/astro/AstronomicalSeason.java#L49

```
    /**
     * Spring equinox for the northern hemisphere in March (or Fall on the southern hemisphere).
     */
    NORTHWARD_EQUINOX,
    /**
     * Summer solstice for the northern hemisphere in June (or Winter on the southern hemisphere).
     */
    NORTHERN_SOLSTICE,
    /**
     * Fall equinox for the northern hemisphere in September (or Spring on the southern hemisphere).
     */
    SOUTHWARD_EQUINOX,
    /**
     * Winter solstice for the northern hemisphere in December (or Summer on the southern hemisphere).
     */
    SOUTHERN_SOLSTICE;
```

Which I used it in the library I maintained https://github.com/persian-calendar/equinox/blob/main/src/main/java/io/github/persiancalendar/Equinox.java because personally, as we don't use Gregorian calendar in my country, Gregorian calendar months are a little confusing for me as a developer while this naming is more clear to me, yet, it uses jargons may are less popular so I'm not sure about the change but I thought I just can suggest it at least. Sure I won't suggest spring/summer/fall/winter as they are untrue for the southern hemisphere which I'm considering even with my app which has a local target but someone wanted southern hemisphere support from it. Thanks!

**Comment by cosinekitty** [comment]
I do like that these names are more international. Unfortunately, this is yet another case of having to make a painful decision to improve names while breaking other people's existing code, or to have inconsistent names in different programming languages. For the equinoxes, it is a little strange to say they belong to the north or the south in particular. I suppose it indicates which hemisphere's days are getting longer?

**Reply by ebraminio** [reply]
> it is a little strange to say they belong to the north or the south in particular

Agreed.

> I suppose it indicates which hemisphere's days are getting longer?

I think so, for me personally this makes sense as northward equinox somehow meant Spring on northern hemisphere and southward equinox meant Spring for southern hemisphere, what I couldn't feel with March and September, they are alien to me (not that we don't know Gregorian calendar but we don't need to memorize its very details like full sequence of its months naming and so but for people specially do international business it is very common to do of course)

> Unfortunately, this is yet another case of having to make a painful decision to improve names while breaking other people's existing code, or to have inconsistent names in different programming languages.

Sure, understandable
